packages/shared/lib/core/layer-2/utils/validateEthereumAddress.ts

```diff
@@ -0,0 +1,41 @@
+import { Keccak } from 'sha3'
+import { get } from 'svelte/store'
+
+import { localize } from '@core/i18n'
+import { networkHrp } from '@core/network'
+import { validateBech32Address } from '@core/utils'
+
+const HASH_SIZE = 256
+
+export function validateEthereumAddress(address: string): string {
+    if (!validateBech32Address(get(networkHrp), address)) {
+        return localize('error.layer2.layer1Recipient')
+    }
+    // Check if it has the basic requirements of an address
+    if (!/^(0x)?[0-9a-f]{40}$/i.test(address)) {
+        return localize('error.send.invalidAddress')
+        // Check if it's all small caps or all large caps
+    } else if (/^(0x)?[0-9a-f]{40}$/.test(address) || /^(0x)?[0-9A-F]{40}$/.test(address)) {
+        return null
+    } else {
+        // Otherwise check EIP-55 checksum
+        return isChecksumAddress(address)
+    }
+}
+
+// Check if EIP-55 mixed-case checksum address encoding is followed (https://eips.ethereum.org/EIPS/eip-55)
+function isChecksumAddress(address: string): string {
+    address = address.replace('0x', '')
```

Comment:
is it possible that  "0x" is also somewhere in the middle of the address?  
if yes we could also use substring(2)

Reply:
No, it's a hex string, we already verify that is according to that format [here](ttps://github.com/iotaledger/firefly/blob/bba9a68235220d1dfad96143e5a07dd01ccea31b/packages/shared/lib/core/layer-2/utils/validateEthereumAddress.ts#L20)